### PR TITLE
make jikeloader can specify statistics type

### PR DESCRIPTION
**Pull request by umm233**
Option argument `count_type`, you can specify statistics type:
- `record`: post num (default)
- `like`: post be liked num
- `share`: post be share num
- `comment`: post be comment num
- `repost`: post be repost num

### CODE REVIEW

**Comment by yihong0618** [comment]
will take a look tomorrow

**Comment by yihong0618** [comment]
and release to pypi

**Comment by umm233** [comment]
> and release to pypi

ok

File: github_poster/loader/jike_loader.py [hunk 1]
```diff
@@ -77,17 +90,19 @@ def _get_first_last_id(self):
         if not r.ok:
             raise LoadError("Can not get first last id, please check your cookie")
         else:
-            return find_last_id_in_html(r.text), find_dateTime_in_html(r.text)
+            return find_last_id_in_html(r.text), find_count_dict_by_type_in_html(
+                r.text, self.count_type
+            )
 
-    def _request_post_data(self, last_id="", resp_date_list=[]):
+    def _request_post_data(self, last_id="", resp_data_list=[]):
```
Comment (yihong0618) [inline github_poster/loader/jike_loader.py:100]: do not use this kind of args  `resp_data_list=[]`
instead 
```python
def _request_post_data(self, last_id="", resp_data_list=None):
    if not resp_data_list:
        resp_data_list = []
```

File: github_poster/loader/jike_loader.py [hunk 2]
```diff
@@ -134,48 +164,68 @@ def _request_post_data(self, last_id="", resp_date_list=[]):
         r = self.session.post(
             JIKE_GRAPHQL_URL, headers=self.headers, data=json.dumps(payload_data)
         )
+        # print(r.json())
```
Comment (yihong0618) [inline github_poster/loader/jike_loader.py:170]: maybe delete ?

File: github_poster/loader/jike_loader.py [hunk 3]
```diff
@@ -134,48 +164,68 @@ def _request_post_data(self, last_id="", resp_date_list=[]):
...
+        return self.stop_time == 3
 
     def get_api_data(self):
-        dateTime_cache = []
-        # get first last id
-        first_last_id, first_dateTime_list = self._get_first_last_id()
+        data_cache = []
+        # get first last id and first data
+        first_last_id, first_data_dict = self._get_first_last_id()
         if first_last_id == "":
```
Comment (yihong0618) [inline github_poster/loader/jike_loader.py:194]: if not first_last_id

File: github_poster/loader/jike_loader.py [hunk 4]
```diff
@@ -134,48 +164,68 @@ def _request_post_data(self, last_id="", resp_date_list=[]):
...
+            data_cache.extend(post_resp_dates)
+            return data_cache, first_data_dict
 
     def make_track_dict(self):
-        data_list = self.get_api_data()
-        for d in data_list:
-            date_str = pendulum.parse(d, tz=self.time_zone).to_date_string()
-            self.number_by_date_dict[date_str] += 1
+        count_types = ("record", "like", "comment", "repost", "share")
+        if self.count_type not in count_types:
```
Comment (yihong0618) [inline github_poster/loader/jike_loader.py:207]: I think we do not check here.
check in `cli` may better?
image

https://docs.python.org/3/library/argparse.html

**Review by yihong0618 — APPROVED** [review]
LGTM maybe some change?

**Comment by umm233** [comment]
> LGTM maybe some change?

I will fix it later，thanks.

**Comment by yihong0618** [comment]
Thanks

**Pull request merged**